build-logic/java-api-extractor/src/test/groovy/org/gradle/internal/tools/api/ApiClassExtractorAnnotationsTest.groovy

```diff
@@ -310,6 +313,75 @@ class ApiClassExtractorAnnotationsTest extends ApiClassExtractorTestSupport {
         annotation.annotationType() == extractedAnn
         def stringValues = annotation.names()
         stringValues == ['foo', 'bar']
+    }
+
+    void "default values for annotations are retained"() {
+        given:
+        def api = toApi([
+            A  : '@Ann public class A {}',
+            Ann: '''
+                import java.lang.annotation.ElementType;
+                import java.lang.annotation.Retention;
+                import java.lang.annotation.RetentionPolicy;
+                import java.lang.annotation.Target;
+
+                @Retention(RetentionPolicy.RUNTIME)
+                @Target({ElementType.TYPE})
+                @interface Ann {
+                    int[] arr() default {};
```

Comment:
🤔 Does it make a difference whether the array has elements in the test?

Reply:
Only empty arrays are allowed as defaults for arrays parameters.